src/node_buffer.cc

```diff
@@ -228,7 +231,9 @@ MaybeLocal<Object> New(Isolate* isolate,
     return scope.Escape(buf);
 
   // Object failed to be created. Clean up resources.
-  free(data);
+  if ( data != nullptr ){
+      free(data);
+  }
```

Comment:
This is a needless check, `data` is compared to `nullptr` above.

-1 on this change.